src/kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs

```diff
@@ -457,8 +459,32 @@ impl<'tcx> GotocCtx<'tcx> {
                     let relative_discr = if *niche_start == 0 {
                         niche_val
                     } else {
-                        // This should be a wrapping sub.
-                        niche_val.sub(Expr::int_constant(*niche_start, discr_ty.clone()))
+                        // Compute "niche_val - niche_start" where "-" is
```

Comment:
nit: can you move this to a different function to make this more readable

Reply:
Done.